net/tcp/tcp_conn.c

```diff
@@ -1313,6 +1317,12 @@ int tcp_connect(FAR struct tcp_conn_s *conn, FAR const struct sockaddr *addr)
   conn->tcpstateflags = TCP_SYN_SENT;
   tcp_initsequence(conn->sndseq);
 
+  /* Save initial sndseq to rexmit_seq, otherwise it will zero */
+
+#if !defined(CONFIG_NET_TCP_WRITE_BUFFERS)
```

Comment:
```suggestion
#if !defined(CONFIG_NET_TCP_WRITE_BUFFERS) || \
    defined(CONFIG_NET_SENDFILE)
```

Reply:
Ditto!
Actually the rexmit_seq is modified in sendfile, but when TCP_WRITE_BUFFERS is enabled its value is not copied to sndseq.